packages/client/hooks/useForm.tsx

```diff
@@ -44,7 +46,7 @@ type FormAction = SetError | SetDirty | SetValue
 const reducer = (state: FieldState<any>, action: FormAction) => {
   switch (action.type) {
     case 'setDirty':
-      if (state[action.name].dirty) return state
+      if (state[action.name]?.dirty) return state
```

Comment:
+2 great work fixing setDirty & friends! 


-1 you can remove all of the conditional access things in this file if you do the following:

```ts
interface SetError<T extends string> {
  type: 'setError'
  name: T
  error: string | undefined
}

interface SetDirty<T extends string> {
  type: 'setDirty'
  name: T
}

interface SetValue<T extends string> {
  type: 'setValue'
  name: T
  value: string
}

type FormAction<T extends string> = SetError<T> | SetDirty<T> | SetValue<T>

...
  const [state, dispatch] = useReducer<Reducer<FieldState<T>, FormAction<FieldStateKey<T>>>>(
```

Reply:
ping on this issue @nickoferrall

Reply:
Thanks, @mattkrick. I was away on Thursday & Friday, so just getting to this now. Update pushed, and I'll now merge!